### PR TITLE
Update admonition docs

### PR DESCRIPTION
**Pull request by mortenpi**
Also implements the post-merge feedback from #33279.

### CODE REVIEW

File: stdlib/Markdown/docs/src/index.md [hunk 1]
```diff
@@ -365,23 +365,26 @@ They can be defined using the following `!!!` syntax:
...
-E.g. `"Note"` in the case of the `note` admonition.
+You can also use your own admonition types, as long as the type name only contains lowercase Latin characters (a-z).
+For example, you could have a `terminology` block like this:
 
-If you would like to define your own block, for example a `terminology`  block
-used like so:
 ```
 !!! terminology "julia vs Julia"
+
     Strictly speaking, Julia refers to the language,
```
Comment (StefanKarpinski) [inline stdlib/Markdown/docs/src/index.md:378]:
```suggestion
    Strictly speaking, "Julia" refers to the language,
```

File: stdlib/Markdown/docs/src/index.md [hunk 2]
```diff
@@ -365,23 +365,26 @@ They can be defined using the following `!!!` syntax:
...
     and julia the standard implementation.
```
Comment (StefanKarpinski) [inline stdlib/Markdown/docs/src/index.md:379]:
```suggestion
    and "julia" the standard implementation.
```

Comment (mortenpi) [reply stdlib/Markdown/docs/src/index.md:379]:
```suggestion
    and julia to the standard implementation.
```

File: stdlib/Markdown/docs/src/index.md [hunk 3]
```diff
@@ -365,23 +365,26 @@ They can be defined using the following `!!!` syntax:
...
 
-If you would like to define your own block, for example a `terminology`  block
-used like so:
 ```
 !!! terminology "julia vs Julia"
-    Strictly speaking, Julia refers to the language,
-    and julia the standard implementation.
+
+    Strictly speaking, "Julia" refers to the language,
+    and "julia" the standard implementation.
```
Comment (mortenpi) [inline stdlib/Markdown/docs/src/index.md:379]:
```suggestion
    and "julia" to the standard implementation.
```

**Comment by mortenpi** [comment]
I forgot to bump this after committing the feedback. I think it's good to go?

**Pull request merged**